server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedIT.java

```diff
@@ -500,6 +500,98 @@ public void testExplain() throws Exception {
+                        .startArray("nested"+i)
+                        .startObject()
+                        .field("n_field"+i, "n_value"+i)
+                        .endObject()
+                        .startObject()
+                        .field("n_field"+i, "n_value"+i)
+                        .endObject()
+                        .endArray()
+                        .endObject()
+                )
+                .setRefreshPolicy(IMMEDIATE)
+                .get();
+        }
+
+        SearchResponse searchResponse = client().prepareSearch("test")
+            .setQuery(nestedQuery("nested1", termQuery("nested1.n_field1", "n_value1"), ScoreMode.Total))
+            .setExplain(true)
+            .get();
+        assertNoFailures(searchResponse);
+        assertThat(searchResponse.getHits().getTotalHits().value, equalTo(1L));
+        Explanation explanation = searchResponse.getHits().getHits()[0].getExplanation();
+        assertThat(explanation.getValue(), equalTo(searchResponse.getHits().getHits()[0].getScore()));
+        assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs in range"));
+    }
+
+    public void testExplainSingleDocIndexingWithDeletedDoc() throws Exception {
+        assertAcked(
+            prepareCreate("test").setMapping(
+                jsonBuilder().startObject()
+                    .startObject("properties")
+                    .startObject("nested1")
+                    .field("type", "nested")
+                    .endObject()
+                    .endObject()
+                    .endObject()
+            )
+        );
+
+        ensureGreen();
+
+        client().prepareIndex("test")
+            .setId("1")
+            .setSource(
+                jsonBuilder().startObject()
+                    .field("field1", "value1")
+                    .startArray("nested1")
+                    .startObject()
+                    .field("n_field1", "n_value1")
+                    .endObject()
+                    .startObject()
+                    .field("n_field1", "n_value1")
+                    .endObject()
+                    .endArray()
+                    .endObject()
+            )
+            .setRefreshPolicy(IMMEDIATE)
+            .get();
+
+        indexRandomForConcurrentSearch("test");
+
+        SearchResponse searchResponse = client().prepareSearch("test")
+            .setQuery(nestedQuery("nested1", termQuery("nested1.n_field1", "n_value1"), ScoreMode.Total))
+            .setExplain(true)
+            .get();
+        assertNoFailures(searchResponse);
+        assertThat(searchResponse.getHits().getTotalHits().value, equalTo(1L));
+        Explanation explanation = searchResponse.getHits().getHits()[0].getExplanation();
+        assertThat(explanation.getValue(), equalTo(searchResponse.getHits().getHits()[0].getScore()));
+        assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs in range"));
```

Comment:
Is there no difference in explain string with and without deleted docs ?

Reply:
yes, there is no other difference apart from the range.